### PR TITLE
Rewrite Autofocus javascript as an ES Module

### PR DESCRIPTION
**Pull request by kr8n3r**
We want to migrate all of our legacy Javascript files to ESM, so this is the latest in the series of conversions.

Rewrite follows the same conventions we previously used for FocusBanner and Collapsible checkboxes, [ColourPreview](https://github.com/alphagov/notifications-admin/pull/5321) and [FileUpload](https://github.com/alphagov/notifications-admin/pull/5323)

Added bonus: removes the use of jQuery as well

In this PR we:

- rewrite the module as ES Module
- update the tests to be ES Module
- remove old module JS file
- remove old module from the legacy bundle build

### CODE REVIEW

File: app/assets/javascripts/esm/autofocus.mjs [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    // if the page loads with a scroll position, we can't assume the item to focus onload
+    // is still where users intend to start
+    if ((window.scrollY > 0) && !forceFocus) { return; }
+
+    // See if the component itself is something we want to send focus to
+    // Otherwise look inside the component to see if there are any elements
+    // we want to send focus to
+    let target = arrayOfTargetElements.includes(this.$component.tagName.toLowerCase())
+      ? this.$component
+      : this.filterByElements(arrayOfTargetElements);
```
Comment (tombye) [inline app/assets/javascripts/esm/autofocus.mjs:30]: Could this just use `querySelector` to search within the component using the part of CSS selectors that let you specify a number of different selector patterns?, like:
```js
this.$component.querySelector(arrayOfTargetElements.join(',')) // this.$component.querySelector('input, textarea, select')
```

File: tests/javascripts/autofocus.test.mjs [hunk 2]
```diff
@@ -1,30 +1,33 @@
...
+    });
+
     // set up DOM
     document.body.innerHTML =
       `<div id="wrapper">
         <label class="form-label" for="search">
           ${labelText}
         </label>
-        <input autocomplete="off" class="form-control form-control-1-1" id="search" name="search" type="search" value="" data-notify-module="autofocus">
+        <input autocomplete="off" class="form-control form-control-1-1" id="search" name="search" type="search" value="" data-notify-module="autofocus" >
```
Comment (tombye) [inline tests/javascripts/autofocus.test.mjs:30]:
```suggestion
        <input autocomplete="off" class="form-control form-control-1-1" id="search" name="search" type="search" value="" data-notify-module="autofocus">
```

**Review by tombye — CHANGES_REQUESTED** [review]
This is great, thanks for doing 🚀 . Only a few changes to request and one it some kind of pedantry.

File: app/assets/javascripts/esm/autofocus.mjs (hunk 1 above)
Comment (kr8n3r) [reply app/assets/javascripts/esm/autofocus.mjs]: oh, very clever!

**Review by tombye — APPROVED** [review]
Works great, still. Thanks for making the changes, now good to go 👍🏻

**Pull request merged**